src/docs/stan-reference/examples.tex

```diff
@@ -7000,12 +7000,10 @@ \section{Simulating from a Gaussian Process}
   real x[N];
 }
 transformed data {
-  vector[N] mu;
-  cov_matrix[N] K;
-  mu = rep_vector(0, N);
-  K = cov_exp_quad(x, 1, 1);
-  for (i in 1:N)
-    K[i, i] = K[i, i] + 0.1;
+  matrix[N, N] K = cov_exp_quad(x, 1.0, 1.0);
```

Comment:
At the end of the transformed data and transformed parameter blocks, all constraints are validated.  If transformed data fails, the algorithm never gets off the ground.  If transformed parameters fail, it results in a rejection.